cs/Compiler.csproj

```diff
@@ -0,0 +1,96 @@
+    <None Include="$(ProjectDir)..\compiler\Main.hs" />
+    <None Include="$(ProjectDir)..\compiler\IO.hs" />
+    <None Include="$(ProjectDir)..\compiler\Options.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Lexer.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Parser.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Util.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Syntax\Internal.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Syntax\JSON.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Syntax\SchemaDef.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Syntax\Types.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Syntax\Util.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\CustomMapping.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\TypeMapping.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Util.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Templates.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\ApplyOverloads.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Apply_cpp.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Apply_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Comm_cpp.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Comm_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Enum_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Grpc_cpp.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Grpc_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Reflection_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Types_cpp.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Types_h.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cpp\Util.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cs\Comm_cs.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cs\Grpc_cs.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cs\Types_cs.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cs\Util.hs" />
+    <None Include="$(ProjectDir)..\compiler\src\Language\Bond\Codegen\Cs\Grpc_cs.hs" />
+  </ItemGroup>
+
+  <Import Project="$(MSBuildThisFileDirectory)\build\internal\Common.Internal.targets" />
+
+  <PropertyGroup>
+    <ExternalBuildDependsOn>
+      CoreExternalBuild;
+    </ExternalBuildDependsOn>
+  </PropertyGroup>
+
+  <Choose>
```

Comment:
I find that Choose makes sense when there are more things to set than just one property. For just this one property, this is shorter:

```
<PropertyGroup>
    <BuildDependsOn>SkipExternalBuild</BuildDependsOn>
    <BuildDependsOn Condition=" '$(OS)' == 'Windows_NT' ">ExternalBuild</BuildDependsOn>
</PropertyGroup>
```